packages/rrweb-snapshot/src/css.ts

```diff
@@ -437,15 +437,87 @@ export function parse(css: string, options: ParserOptions = {}) {
     }
     /* @fix Remove all comments from selectors
      * http://ostermiller.org/findcomment.html */
-    return trim(m[0])
+    const splitSelectors = trim(m[0])
       .replace(/\/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*\/+/g, '')
       .replace(/"(?:\\"|[^"])*"|'(?:\\'|[^'])*'/g, (m) => {
         return m.replace(/,/g, '\u200C');
       })
-      .split(/\s*(?![^(]*\)),\s*/)
-      .map((s) => {
+      .split(/\s*(?![^(]*\)),\s*/);
+
+    if (splitSelectors.length <= 1) {
+      return splitSelectors.map((s) => {
         return s.replace(/\u200C/g, ',');
       });
+    }
+
+    // For each selector, need to check if we properly split on `,`
+    // Example case where selector is:
+    // .bar:has(input:is(:disabled), button:is(:disabled))
+    let i = 0;
+    let j = 0;
+    const len = splitSelectors.length;
+    const finalSelectors = [];
+    while (i < len) {
+      // Look for selectors with opening parens - `(` and search rest of
+      // selectors for the first one with matching number of closing
+      // parens `)`
+      const openingParensCount = (splitSelectors[i].match(/\(/g) || []).length;
+
+      if (openingParensCount > 1) {
+        // At least one opening parens was found, prepare to look through
+        // rest of selectors
```

Comment:
comment suggests that it should be >=

```suggestion
      if (openingParensCount >= 1) {
        // At least one opening parens was found, prepare to look through
        // rest of selectors
```

This is what got me thinking of a test with one set of braces, and comma in the middle.